From 6e3ee52e726a6ad118be5b31c8deff24b7f84262 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 14 Jul 2020 11:42:51 +0700
Subject: [PATCH 068/440] gpio: rcar: Prevent to force the IRQ handler as
 threaded on RT system

When kernel is booting with CONFIG_IRQ_FORCED_THREADING flag, the
gpio_rcar_irq_handler() will be forced as threaded and then will
generate a WARNING relating to interrupt handlers. To prevent to
force the interrupt handler as threaded, we will add IRQF_NO_THREAD
flag when requesting an interrupt for gpio-rcar driver.

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 drivers/gpio/gpio-rcar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-rcar.c b/drivers/gpio/gpio-rcar.c
index 58deb20..e1b5914 100644
--- a/drivers/gpio/gpio-rcar.c
+++ b/drivers/gpio/gpio-rcar.c
@@ -560,7 +560,7 @@ static int gpio_rcar_probe(struct platform_device *pdev)
 
 	p->irq_parent = irq->start;
 	if (devm_request_irq(dev, irq->start, gpio_rcar_irq_handler,
-			     IRQF_SHARED, name, p)) {
+			IRQF_SHARED | IRQF_NO_THREAD, name, p)) {
 		dev_err(dev, "failed to request IRQ\n");
 		ret = -ENOENT;
 		goto err1;
-- 
2.7.4

